SimG4CMS/ShowerLibraryProducer/src/FiberSD.cc

```diff
@@ -19,12 +22,14 @@
 #include "G4ios.hh"
 
 FiberSD::FiberSD(const std::string& iname,
-                 const DDCompactView& cpv,
+                 const edm::EventSetup& es,
                  const SensitiveDetectorCatalog& clg,
                  edm::ParameterSet const& p,
                  const SimTrackManager* manager)
-    : SensitiveCaloDetector(iname, cpv, clg, p), m_trackManager(manager), theHCID(-1), theHC(nullptr) {
-  theShower = new HFShower(iname, cpv, p, 1);
+    : SensitiveCaloDetector(iname, es, clg, p), m_trackManager(manager), theHCID(-1), theHC(nullptr) {
+  edm::ESTransientHandle<DDCompactView> cpv;
+  es.get<IdealGeometryRecord>().get(cpv);
+  theShower = new HFShower(iname, *cpv, p, 1);
```

Comment:
@ianna , here we may send EventSetup instead of DDCompactView ?

Reply:
@civanch - the goal of this PR is to decouple DD dependencies for SD classes and identify which one do not need to be changed. I'd like to see an impact before going into more details.